pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,376 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from "react";
+import PropTypes from 'prop-types';
+import {
+    Alert, Button, Checkbox,
+    Flex, FlexItem,
+    Modal, Radio,
+    Split, SplitItem,
+    Stack,
+    Text, TextVariants
+} from '@patternfly/react-core';
+import { InfoIcon } from "@patternfly/react-icons";
+
+import cockpit from "cockpit";
+import { proxy as serviceProxy } from 'service';
+import { check_missing_packages } from "packagekit.js";
+import { install_dialog } from "cockpit-components-install-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+export class KpatchSettings extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            auto: null,
+            loaded: false,
+            enabled: null,
+            missing: [],
+            unavailable: [],
+            error: "",
+            updating: false,
+            showModal: false,
+            applyCheckbox: false,
+            justCurrent: null,
+            kernelName: "",
+            patchName: null,
+            patchInstalled: null,
+            patchUnavailable: null,
+        };
+
+        this.kpatchService = serviceProxy("kpatch");
+
+        this.checkSetup = this.checkSetup.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.onClose = this.onClose.bind(this);
+        this.handleInstall = this.handleInstall.bind(this);
+    }
+
+    componentDidMount() {
+        check_missing_packages(["kpatch", "kpatch-dnf"])
+                .then(d => {
+                    this.setState({
+                        loaded: true,
+                        unavailable: d.unavailable_names || [],
+                        missing: d.missing_names || [],
+                    });
+
+                    this.checkSetup();
+                })
+                .catch(e => {
+                    console.log("Could not determine kpatch availability.");
```

Comment:
Let's print the error message as well for easier debugging. Also you can drop the { } wrapper.